src/utils.jl

```diff
@@ -284,23 +284,37 @@ end
 `combinedef` is the inverse of `splitdef`. It takes a splitdef-like Dict
 and returns a function definition. """
 function combinedef(dict::Dict)
-  rtype = get(dict, :rtype, :Any)
+  rtype = get(dict, :rtype, nothing)
   params = get(dict, :params, [])
   wparams = get(dict, :whereparams, [])
   name = dict[:name]
   name_param = isempty(params) ? name : :($name{$(params...)})
   # We need the `if` to handle parametric inner/outer constructors like
   # SomeType{X}(x::X) where X = SomeType{X}(x, x+2)
   if isempty(wparams)
-    :(function $name_param($(dict[:args]...);
-                           $(dict[:kwargs]...))::$rtype
-      $(dict[:body])
-      end)
+    if rtype==nothing
+      @q(function $name_param($(dict[:args]...);
+                              $(dict[:kwargs]...))
+        $(dict[:body].args...)
```

Comment:
We should probably figure out why the `begin` is introduced in the first place. Maybe it's because I used `longdef1(fdef)` in `splitdef`. Perhaps changing it to `shortdef1` (with associated changes) would fix the issue.